packages/picasso/src/utils/Modal/modal-manager.ts

```diff
@@ -17,11 +21,38 @@ export class ModalManager {
     }
 
     this.modals.splice(modalIndex, 1)
+    this.liftScrollLock()
   }
 
   isTopModal(modalId: number) {
     return (
       this.modals.length > 0 && this.modals[this.modals.length - 1] === modalId
     )
   }
+
+  private dropScrollLock() {
+    if (!isBrowser()) {
+      return
+    }
+
+    if (!isDefined(this.prevBodyOverflow)) {
+      this.prevBodyOverflow = window.getComputedStyle(document.body).overflow
+      document.body.style.overflow = 'hidden'
+    }
+  }
+
+  private liftScrollLock() {
+    if (!isBrowser()) {
+      return
+    }
+
+    if (isDefined(this.prevBodyOverflow) && this.modals.length === 0) {
+      document.body.style.overflow = this.prevBodyOverflow
+      this.prevBodyOverflow = undefined
+    }
+  }
+}
+
+const isDefined = function <T>(val: T | undefined): val is T {
```

Comment:
<img width="828" alt="image" src="https://user-images.githubusercontent.com/903437/174582176-92c8d3b3-c40c-4c66-97ab-74b5989500e3.png">

btw, there's a conflict between eslint and prettier configuration. eslint wants a space before parenthesis (and reports an error, not just warning), while prettier always removes that space

IMO it would make sense to remove this rule from eslintrc at all because I as a developer don't want to care how it's formatted as prettier will always do it same way https://github.com/toptal/davinci/blob/master/packages/syntax/src/configs/.eslintrc#L42-L49

Reply:
I would recommend using arrow functions in this case:
```suggestion
const isDefined = <T,>(val: T | undefined): val is T => {
```

About the problem with prettier, we have a edge case with anonymous functions and generic arguments, I will try to address it

Reply:
For some reason I was sure that generic arrow functions are not possible, but now it works just fine. Probably it was just older TS version when I last tried, or some specific eslint config...

Anyway thanks for suggestion!